tests/client_test.go

```diff
@@ -489,6 +489,17 @@ func TestStats(t *testing.T) {
 	if err != nil {
 		t.Errorf("Couldn't remove servers: %v", err)
 	}
+
+	locationZones := stats.LocationZones
```

Comment:
To write meaningful unit tests here you'll have to add a `server_zone` to a location block in `test.conf`

At the moment an empty map is returned. Checking for nil values doesn't ensure stats were populated.

Reply:
yep, we need to at least check that we have the right number of zones returned. You can add a zone to the /test location of the test.conf.

Reply:
This test should happen before `_, _, err = c.UpdateHTTPServers(upstream, []client.UpstreamServer{})` just above. 

The upstream cleanup/deletion should always be the last step of the function.

Reply:
Can this test be improved? Something like
```go
if locZones, ok := stats.LocationZones[upstream]; ok {
	if locZones.Requests < 1 {
		t.Errorf("LocationZone stats missing: %v", locZones.Requests)
	}
} else {
	t.Errorf("LocationZone %v not found", upstream)
}
```

**Note** This will also require investigating how to generate non 0 stats for location zones.